spynnaker/pyNN/models/neuron/synaptic_manager.py

```diff
@@ -799,6 +799,9 @@ def write_data_spec(
 
         self._weight_scales[placement] = weight_scales
 
+    def clear_connection_cache(self):
+        self._retrieved_blocks = dict()
```

Comment:
i imagine that structural plastic can change static synapses as well. peter would be best positioned to figure this out. if this is simple to implement, id say its a post optimisation over this fix. i wouldn't say its a blocker to this fix itself. 

i assume youd need to delete the entire block instead of just the synapse row, as then the caching mechanism is going to be much more complicated. This idea would also be safe in pynn 0.8 where the projection gives you access to all params, as the static is only weight and delay, which unless structual effects these, would be constant. 

worth noting that this may interact with post run set of a projection weight (which we will need to support at some point)